### PR TITLE
[fix](group commit) Fix group commit VOlapTablePartitionParam memory …

### PR DESCRIPTION
**Pull request by mymeiyi**
…leak

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/vec/sink/group_commit_block_sink.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ Status GroupCommitBlockSink::init(const TDataSink& t_sink) {
     _group_commit_mode = table_sink.group_commit_mode;
     _load_id = table_sink.load_id;
     _max_filter_ratio = table_sink.max_filter_ratio;
-    _vpartition = new doris::VOlapTablePartitionParam(_schema, table_sink.partition);
+    _vpartition = std::make_shared<doris::VOlapTablePartitionParam>(_schema, table_sink.partition);
```
Comment (gavinchou) [inline be/src/vec/sink/group_commit_block_sink.cpp:62]: why not `make_unique<>`?

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Pull request merged**